src/plugins/content_management/server/core/core.ts

```diff
@@ -53,4 +61,15 @@ export class Core {
       },
     };
   }
+
+  private setupEventStream() {
+    // TODO: This should be cleaned up and support added for all CRUD events.
```

Comment:
It could be, that can be done in the [follow up](https://github.com/orgs/elastic/projects/1102/views/21?pane=issue&itemId=22914805), if we decide so. I was thinking to do so as well, but then decided (at least for now) to do everything inside the CM `Core` as then the CM `Core` still remains the single root object of CM. I.e.:

```ts
const core = new Core(logger, eventStream);
```

If it is done somewhere outside of `Core`, then that other thing (`init` in your example) conceptually becomes a higher-level thing:

```ts
init(core, eventStream)
```

Like:

```ts
const core2 = new Core2(core, eventStream);

core2.init();
```

So we conceptually have then this other `core2` service, which is parent of the current `core`. Which is not bad, I just thought it is simpler to keep the current setup for now.